http/tus_handlers.go

```diff
@@ -148,12 +208,26 @@ func tusPatchHandler() handleFunc {
 			return http.StatusInternalServerError, fmt.Errorf("could not write to file: %w", err)
 		}
 
-		w.Header().Set("Upload-Offset", strconv.FormatInt(uploadOffset+bytesWritten, 10))
+		newOffset := uploadOffset + bytesWritten
+		w.Header().Set("Upload-Offset", strconv.FormatInt(newOffset, 10))
+
+		if newOffset >= uploadLength {
+			unregisterUpload(file.RealPath())
```

Comment:
This is the only place where we call `unregisterUpload`. What happens if the upload fails/is canceled?

Reply:
Good point, i forgot to handle uploads that were gracelly terminated the frontend. I can implement `tusDeleteHandler `to handle the case where the frontend cancels the upload.

In the event of an upload failure, I assume the user would retry, making it to naturally unregister, effectively avoiding any negative impact. Properly handling upload failures would require implementing some form of background processing, given the limited benefit in our current context, I’ve decided to skip it for now. However, if desirable, i can give it a try. One approach I’ve considered is using goroutines to count down the timeout.

Reply:
Good point, i forgot to handle uploads that were gracelly terminated the frontend. I can implement `tusDeleteHandler` to handle the case where the frontend cancels the upload.

In the event of an upload failure, I assume the user would retry, making it to naturally unregister, effectively avoiding any negative impact. Properly handling upload failures would require implementing some form of background processing. Given the limited benefit in our current context and seeing that even the current implementation doesn't handle failures, I’ve decided to skip it for now.

Reply:
A possibility would be to use some sort of caching like [otter](https://pkg.go.dev/github.com/maypok86/otter/v2@v2.2.0), with automatic deletion of entries if there haven't been any update after _n_ time

Reply:
It looks like otter is what we need, but it requires Go version 1.24, whereas the current project is using 1.23. What do you thing about using an alternative like [ristretto](https://pkg.go.dev/github.com/dgraph-io/ristretto) or [ttlcache](https://pkg.go.dev/github.com/jellydator/ttlcache) which, by the way, appear to be more widely adopted?

Reply:
That's also fine.